cudamapper/src/cudamapper_utils.cpp

```diff
@@ -65,5 +66,63 @@ void fuse_overlaps(std::vector<Overlap>& fused_overlaps, const std::vector<Overl
     set_relative_strand(fused_overlap);
     fused_overlaps.push_back(fused_overlap);
 }
+
+std::string string_slice(const std::string& s, std::size_t start, std::size_t end)
+{
+    return s.substr(start, end - start);
+}
+
+std::vector<std::string> kmerize_string(const std::string& s, int kmer_length, int stride)
+{
+    std::size_t kmer_count = s.length() - kmer_length + 1;
+    std::vector<std::string> kmers;
+    for (std::size_t i = 0; i < kmer_count; i += stride)
+    {
+        kmers.push_back(s.substr(i, i + kmer_length));
+    }
+    return kmers;
+}
+
+template <typename T>
+std::size_t count_shared_elements(const std::vector<T>& a, const std::vector<T>& b)
+{
+    std::size_t a_index      = 0;
+    std::size_t b_index      = 0;
+    std::size_t shared_count = 0;
+
+    while (a_index < a.size() && b_index < b.size())
```

Comment:
This method is O(n), but requires sorted vectors which is O(n*log(n)).

To repeat my question from the header file:
```Let's say that vector a has m elements with value x and vector b n such elements. Should this results in 1 or m*n shared elements? I would assume former but your implementation would actually return min(m, n)```

Depending on the answer to that question you could use `std::unordered_set` or `std::unordered_map` with a counter as value to to this on unsorted vectors

Reply:
I'll have to talk with @vellamike about how this might behave in low complexity genomic regions, but I think we want the "counting" behavior that an `unordered_map` would offer. We may have to test both to see.

Reply:
Should I consider this blocking for this PR? I've created a new branch that will address some of the performance issues, including moving to string_view, refactoring the main loop portion, and implementing a counting `unordered_map` approach.

Reply:
I would say we don't have to address this right now

Reply:
I assume it would be invalidated by moving the function to device as well, correct?